src/solvers/eigen_sparse_linear_solver.C

```diff
@@ -101,47 +101,109 @@ EigenSparseLinearSolver<T>::solve (SparseMatrix<T> & matrix_in,
 
   std::pair<unsigned int, Real> retval(0,0.);
 
-  const int max_its = this->get_int_solver_setting("max_its", m_its);
-  const double abs_tol = this->get_real_solver_setting("abs_tol", tol);
+  // Eigen doesn't give us a solver base class?  We'll just use a
+  // generic lambda, then.
+  auto do_solve = [this, &rhs, &solution, tol, m_its]
+    (auto & e_solver, std::string_view msg) {
+    const int max_its = this->get_int_solver_setting("max_its", m_its);
+    const double abs_tol = this->get_real_solver_setting("abs_tol", tol);
```

Comment:
Do these need to be hard-typed or can we use `auto` here?

Reply:
It does seem a little surprising to me that `get_real_solver_setting` returns a `double` instead of a `Real`

Reply:
Heh; I want to blame @grmnptr for that, in #3625, but both John *and* I reviewed that one and neither of us noticed the poor name choice.  (name choice, not return value choice, IMHO, since we generally use `double`-typed tolerances regardless of `Real` type for a few reasons).

That's a new enough change and this is an internal enough API that I think we can just change the name.  I'll want to do that in a separate PR though.

Reply:
I prefer the hard type here over auto; requires less effort to read and understand.  I use auto when a type is way too verbose, or when it's a dependent type in a generic function, or when I'm too lazy to remember exactly which type I want.

Reply:
Yea I think the primary argument for `auto` here would be if you were going to go change the return type of either of these functions, and then we would be doing a conversion perhaps without knowing it. But then again you could be doing a conversion when passing these arguments in the following lines so 🤷